src/ClientResolver.php

```diff
@@ -412,6 +422,21 @@ public static function _apply_debug($value, array &$args, HandlerList $list)
         }
     }
 
+    public static function _apply_stats($value, array &$args, HandlerList $list)
+    {
+        $defaults = array_fill_keys(
```

Comment:
Ohhh, I see what you're doing here. Since it's really clever code, please add a comment to describe what's happening here. I also don't see any explicit tests for this.